### ISSUE TITLE
Linter to enforce docs style guide

### ISSUE THREAD

**Issue by frankfarzan** [kptdev/kpt#1652]
We're going to automatically enforce a style guide on all documentation (reference, book, catalog). This will be a growing list of rules, starting with:

- [x] All references to the tool should be `kpt` (small case).
- [x] No broken links (I believed this is enforced already)
- [ ] Banned words: hydration -> render
- [ ] Function images must be versioned, disallow `unstable` tag
- [ ] All verbatim shell quote blocks should start with `$`. Right: `$ kpt pkg get`  Wrong: `kpt pkg get`
- [ ] Links to other kpt.dev pages should be relative
- [ ] Detect TODO(s)

**Comment by aravindtga** [comment]
Triaged and addressed all items in this issue:

  | Rule | Status | Action | Automated? | Reason |
  |------|--------|--------|-----------|--------|
  | Banned words: hydration → render | Fixed | 1 occurrence replaced manually | No | Used once in 4 years, unlikely to recur |
  | Function images versioned, disallow `:unstable` | No violations | Zero occurrences found | No | Nothing to enforce |
  | All shell blocks should start with `$` | Fixed (reinterpreted) | Removed `$` from command-only blocks, kept in command+output blocks. Added style guide rule 13 | No | Context-dependent - requires knowing if block has output |
  | Links to other kpt.dev pages should be relative | Fixed | Replaced 13 hardcoded `https://kpt.dev/` URLs with rooted paths | No | One-time fix, not an ongoing problem |
  | Detect TODOs | No violations | Only 2 exist, both intentional code comments in examples | No | False positives in code examples |
  | Spelling | Automated | Added `make verify` using `golangci/misspell@v0.8.0`, integrated into `docs.yml` CI workflow. Fixed 5 misspellings | Yes | Ongoing value, same tool golangci-lint uses for `.go` files |

  Addressed in #4690. 

Would appreciate opinions on whether this is sufficient to close the issue or if further automation is desired for any specific rule.

**Comment by Catalin-Stratulat-Ericsson** [comment]
i agree with everything @aravindtga said and modified except for one aspect. 

1. firstly im confused as to why kpt doesnt use relref links like porch does. it would catch these broken relative path at preview deployment time for a PR?
